### PR TITLE
docs(launch):  Add readmes to launch 

### PR DESCRIPTION
**Pull request by iveksl2**
Fixes WB-NNNN
Fixes #NNNN

Description
-----------
What does the PR do?

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

**Comment by iveksl2** [comment]
@bcsherma let me know what you think. 

P.S. @KyleGoyette i think this requires your approval. This is NOT URGENT. I am looking into the code to further my own technical depth and understanding. (PMs should have a good technical foundation to make better product decisions)  I thought creating README's might be a way to pay it forward for the next person 🤷

File: wandb/sdk/launch/agent/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+An agent is broadly defined as a *"program that collects information or performs a task in the background at a particular schedule. The term agent is often thought of as a software abstraction that is capable of acting with a certain degree of autonomy to perform a particular task on behalf of its host"* ([Source](https://www.techopedia.com/definition/1292/agent#:~:text=Techopedia%20Explains%20Agent-,What%20Does%20Agent%20Mean%3F,on%20behalf%20of%20its%20host.))
```
Comment (gtarpenning) [inline wandb/sdk/launch/agent/README.md:1]: do we need this predicate stuff? Kinda fun but are techies really going to spend the time reading? Might be annoying idk

File: wandb/sdk/launch/agent/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+Within the Launch project, an agent polls a run queue and launches & executes runs for wandb launch
```
Comment (gtarpenning) [inline wandb/sdk/launch/agent/README.md:3]:
```suggestion
Within Launch, an agent continually polls a queue of jobs, executing runs on the specified resource that then seamlessly report back to wandb
```

File: wandb/sdk/launch/builder/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+The builder will build the docker image for execution of the run. 
+If you pass a prebuilt docker image, the builder does not need to do anything.
```
Comment (gtarpenning) [inline wandb/sdk/launch/builder/README.md:2]:
```suggestion
If necessary, the builder creates a docker image for execution of the run. 
Prebuilt docker images can be used to skip this step. 
```

**Review by gtarpenning — APPROVED** [review]

File: wandb/sdk/launch/agent/README.md (hunk 1 above)
Comment (iveksl2) [reply wandb/sdk/launch/agent/README.md:1]: Under this umbrella definition the launch agent, sweeps agent & an RL agent, all work 😃

**Pull request merged**